### PR TITLE
Fix for 416 exception when file size is of zero bytes.

### PR DESCRIPTION
**Pull request by sumit-ql**
Fix for issue [6572]. 
1. This issue is reported in Storage, but its basically resumable media issue.
2. Implemented the fix for 416 exception only. Other suggestion can be taken up in seperate PR.

### CODE REVIEW

**Comment by sumit-ql** [comment]
@mf2199 @@sangramql Please review.

File: google/resumable_media/_download.py [hunk 1]
```diff
@@ -466,3 +472,28 @@ def get_range_info(response, get_headers, callback=_helpers.do_nothing):
         int(match.group(u'end_byte')),
         int(match.group(u'total_bytes'))
     )
+
+
+def check_for_zero_content_range(response, get_status_code, get_headers):
+    """ Validate if response status code is 416 and content range is zero.
+
+    This is the speacial case for handling zero bytes files.
```
Comment (mf2199) [inline google/resumable_media/_download.py]: Fix the grammar and typo:
`This is special case for handling zero bytes files.`

**Comment by mf2199** [comment]
As I've had recently stumbled upon, `nox -e lint` may not like some of the formatting. You might want to try `nox -e blacken` to make sure, if not already done so.

File: google/resumable_media/_download.py [hunk 2]
```diff
@@ -466,3 +472,28 @@ def get_range_info(response, get_headers, callback=_helpers.do_nothing):
         int(match.group(u'end_byte')),
         int(match.group(u'total_bytes'))
     )
+
+
+def check_for_zero_content_range(response, get_status_code, get_headers):
```
Comment (mf2199) [inline google/resumable_media/_download.py]: This method should probably be made private. Also, it it is used only once, then encapsulation is not necessary, and the check can be placed inside the `_process_response()` method.

**Review by mf2199 — CHANGES_REQUESTED** [review]

**Comment by sumit-ql** [comment]
> As I've had recently stumbled upon, `nox -e lint` may not like some of the formatting. You might want to try `nox -e blacken` to make sure, if not already done so.

Thanks, yes i will surely try this.

File: google/resumable_media/_download.py (hunk 1 above)
Comment (sumit-ql) [reply google/resumable_media/_download.py]: My bad, will do it.

File: google/resumable_media/_download.py (hunk 2 above)
Comment (sumit-ql) [reply google/resumable_media/_download.py]: Yeah, making this function as a private function make sense. 
I agree, it is currently used once but we may need it somewhere in future. Also, making it as a separate function, will make _process_response() method more cleaner and shorter. Also, it helps to write clean and separate tests for it.

File: google/resumable_media/_download.py (hunk 1 above)
Comment (sumit-ql) [reply google/resumable_media/_download.py]: Done.

File: google/resumable_media/_download.py (hunk 2 above)
Comment (mf2199) [reply google/resumable_media/_download.py]: I don't disagree with this but let's see what Googlers have to say.

**Review by mf2199 — APPROVED** [review]

**Pull request closed without merging**